### ISSUE TITLE
SQLCHECK & SSPICLIENT - Feature Request - Sanity check @@SERVERNAME is equivalent to the requested name

### ISSUE THREAD

**Issue by jamesaepp** [microsoft/CSS_SQL_Networking_Tools#44]
I have been banging my head against the wall for hours trying to figure out why SSPI connections to MS SQL servers suddenly started to fail.

From a blind guess troubleshooting from a completely different direction I found out that the @@SERVERNAME property of the MS SQL server/services/instance was different than what was being contacted in the client.

Knowing this would have saved me a lot of pain and it's something I believe the tools should be able to figure out. At the very least SQLCHECK -- SSPICLIENT may be a different story.

If you need clarification please let me know.

**Comment by Malcolm-Stewart** [comment]
Hi James, SQLCheck gathers static and does not connect to SQL Server in any way. However, this is a candidate for SSPIClient, which does make a connection to SQL Server. Good idea.

Just to follow-up, was the failure due to a SQL Alias or a hosts file entry, both of which will show in the SQLCheck report, or was it due to a DNS CNAME record, which will not show?

Thanks

**Comment by jamesaepp** [comment]
@Malcolm-Stewart :

Surely this would be possible to implement on SQLCHECK? If I run SQLCHECK locally on the DB server (or even SSMS locally) on the DB server everything is operational so I think it would be trivial to check the status of `@@SERVERNAME` and verify it against the computer names & SPNs of the computer object in AD.

As for the issue, it had nothing to do with hosts file, DNS, or aliases. It was simply that when the DB server was put into production and renamed in Windows from SERVERNAME-STAGED to SERVERNAME-PROD , the below steps weren't taken.

https://learn.microsoft.com/en-us/sql/database-engine/install-windows/rename-a-computer-that-hosts-a-stand-alone-instance-of-sql-server?view=sql-server-ver16

**Comment by Malcolm-Stewart** [comment]
Good point. That could be a candidate for SSPIClient. SQLCheck does not make any connections to SQL Server at all.
